task/07-yield-tasks.js

```diff
@@ -126,9 +159,31 @@ function* breadthTraversalTree(root) {
  *   [ 1, 3, 5, ... ], [ -1 ] => [ -1, 1, 3, 5, ...]
  */
 function* mergeSortedSequences(source1, source2) {
-    throw new Error('Not implemented');
+    let s1 = source1(),
+        s2 = source2();
+    let first = s1.next().value,
+        second = s2.next().value;
+    while (first != undefined || second != undefined) {
+        if (first != undefined && second != undefined) {
+            if (first < second) {
+                yield first;
+                first = s1.next().value;
+            } else {
+                yield second;
+                second = s2.next().value;
+            }
+        } else if (first != undefined) {
+            yield first;
+            first = s1.next().value;
+        } else if (second != undefined) {
```

Comment:
"operator !== is always preferable
sometimes ''!=' may cause an error
(check at console :
'0' != 0
and '0' !== 0)"